src/nl/hannahsten/texifyidea/util/magic/CommandMagic.kt

```diff
@@ -4,90 +4,81 @@ package nl.hannahsten.texifyidea.util.magic
 
 import com.intellij.ui.Gray
 import nl.hannahsten.texifyidea.lang.LSemanticCommand
+import nl.hannahsten.texifyidea.lang.LatexContexts
+import nl.hannahsten.texifyidea.lang.LatexLib
 import nl.hannahsten.texifyidea.lang.LatexPackage
-import nl.hannahsten.texifyidea.lang.commands.*
 import nl.hannahsten.texifyidea.lang.commands.LatexBiblatexCommand.*
-import nl.hannahsten.texifyidea.lang.commands.LatexGenericMathCommand.*
 import nl.hannahsten.texifyidea.lang.commands.LatexGenericRegularCommand.*
 import nl.hannahsten.texifyidea.lang.commands.LatexGlossariesCommand.*
-import nl.hannahsten.texifyidea.lang.commands.LatexListingCommand.*
 import nl.hannahsten.texifyidea.lang.commands.LatexMathtoolsRegularCommand.*
 import nl.hannahsten.texifyidea.lang.commands.LatexNatbibCommand.*
 import nl.hannahsten.texifyidea.lang.commands.LatexNewDefinitionCommand.*
-import nl.hannahsten.texifyidea.lang.commands.LatexOperatorCommand.*
-import nl.hannahsten.texifyidea.lang.commands.LatexUncategorizedStmaryrdSymbols.BIG_SQUARE_CAP
 import nl.hannahsten.texifyidea.lang.commands.LatexXparseCommand.*
 import nl.hannahsten.texifyidea.lang.predefined.PredefinedCmdMathSymbols
-import nl.hannahsten.texifyidea.util.magic.CommandMagic.stylePrimitives
+import nl.hannahsten.texifyidea.lang.predefined.AllPredefined
+import nl.hannahsten.texifyidea.lang.predefined.PredefinedCmdFiles
+import nl.hannahsten.texifyidea.lang.predefined.PredefinedCmdGeneric
 
 object CommandMagic {
 
     /**
      * LaTeX commands that make the text take up more vertical space.
      */
     val high = hashSetOf(
-        FRAC.cmd, DFRAC.cmd, SQUARE_ROOT.cmd, SUM.cmd, INTEGRAL.cmd, DOUBLE_INTEGRAL.cmd, TRIPLE_INTEGRAL.cmd,
-        QUADRUPLE_INTEGRAL.cmd, N_ARY_PRODUCT.cmd, N_ARY_UNION.cmd, N_ARY_INTERSECTION.cmd,
-        N_ARY_SQUARE_UNION.cmd, BIG_SQUARE_CAP.cmd
+        "\\frac", "\\dfrac", "\\sqrt", "\\sum", "\\int", "\\iint", "\\iiint", "\\iiiint",
```

Comment:
I saw you changing back and forth in various places between strings and references to (now) CommandNames. Defining them in a single place has been useful to find usages of particular commands, but I can imagine it is annoying to create new entries in CommandNames. What do you think, stick to plain strings?

Reply:
The main reason not everything was migrated to LatexCommand is that when adding commands, I would need to check manually what exactly the parameters are, which package it's from, etc. But now with CommandNames, we could do automatic migration to there like you did, and decide later whether to add those commands to PredefinedCmd*